lib/ddtrace/contrib/rails/core_extensions.rb

```diff
@@ -163,49 +163,36 @@ def patch_process_action
         else
           # Rewrite module that gets composed into the Rails controller base class
           ::ActionController::Instrumentation.class_eval do
-            def process_action_with_datadog(*args)
-              # mutable payload with a tracing context that is used in two different
-              # signals; it propagates the request span so that it can be finished
-              # no matter what
-              payload = {
-                controller: self.class,
-                action: action_name,
-                headers: {
-                  # The exception this controller was given in the request,
-                  # which is typical if the controller is configured to handle exceptions.
-                  request_exception: request.headers['action_dispatch.exception']
-                },
-                tracing_context: {}
-              }
-
-              begin
-                # process and catch request exceptions
-                Datadog::Contrib::Rails::ActionController.start_processing(payload)
-                result = process_action_without_datadog(*args)
-                payload[:status] = response.status
-                result
-              rescue Exception => e
-                payload[:exception] = [e.class.name, e.message]
-                payload[:exception_object] = e
-                raise e
-              end
-            ensure
-              Datadog::Contrib::Rails::ActionController.finish_processing(payload)
-            end
-
             alias_method :process_action_without_datadog, :process_action
-            alias_method :process_action, :process_action_with_datadog
+            remove_method :process_action
+
+            include ActionControllerPatch
```

Comment:
I don't think you can just `include` the patch here; this is something I spent a lot of time trying to figure out. I think you need `prepend` in order to inject the patch at the top of the ancestor stack.

Reply:
`prepend` is only required to override existing methods. In ruby < 2.0 we can still  do that but it requires some additional steps. 

This is to make sure we reuse the same class and instrumentation implementation for both versions. Ensuring there are no difference in actual instrumentation implementation when  running ruby 1.9.3.